src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs

```diff
@@ -58,7 +58,7 @@ public static IConfig Create(
                 .AddValidator(new UniqueArgumentsValidator()) // don't allow for duplicated arguments #404
                 .WithSummaryStyle(SummaryStyle.Default.WithMaxParameterColumnWidth(36)); // the default is 20 and trims too aggressively some benchmark results
 
-            if (Reporter.CreateReporter().InLab)
+            if (new EnvironmentProvider().IsLabEnvironment())
```

Comment:
Could we just make this a static method we call on EnvironmentProvider? Save the allocation.

Reply:
I wanted to have the possibility to mock environment for unit testing - thus the interface and it's implementation. 
At the same time I wanted to refactor only a limited portion code - thus the least possible changes outside the Reporting solution (like here in BenchmarkDotNet.Extensions). However good point with the allocation. I introduced a new static variable of the IEnvironment so the EnvironmentProvider is created only once.